Adding "connection-name" optional parameter

bluedenim opened the pull request:
Updating on poetry installation steps, README.md, and requirements to support Django 1.11+

Inspired by https://github.com/bluedenim/vmigration-helper/pull/14

bluedenim committed:
Adding "connection-name" optional parameter
Updating on poetry installation steps, README.md, and requirements to support Django 1.11+

bluedenim commented on Dockerfile:
```diff
@@ -1,9 +1,9 @@
 # Install and setup Poetry
-RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
-RUN echo export PATH=\$PATH:\$HOME/.poetry/bin >> /root/.bashrc
+RUN curl -sSL https://install.python-poetry.org | python3 -
```
Let's hope their install script is stable now.

setu4993 replied:
😅. They say it's going to be 🤞🏽.

bluedenim commented on vmigration_helper/helpers/command.py:
```diff
@@ -0,0 +1,14 @@
+from django.db import DEFAULT_DB_ALIAS
+
+
+class MigrationCommand(BaseCommand, ABC):
```
Common options for all our migration commands

setu4993 replied:
Nice clean-up of the dupe from #14.

setu4993 commented on vmigration_helper/helpers/command.py:
```diff
@@ -0,0 +1,14 @@
+from abc import ABC
+
+from django.core.management import BaseCommand
+from django.db import DEFAULT_DB_ALIAS
```
```suggestion
from django.conf import settings
```

setu4993 commented on vmigration_helper/helpers/command.py:
```diff
@@ -0,0 +1,14 @@
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--connection-name',
+            default=DEFAULT_DB_ALIAS,
```
```suggestion
            default=settings.DEFAULT_DB_ALIAS,
```

setu4993 commented on requirements.txt:
```diff
@@ -1,3 +1,3 @@
 asgiref==3.4.1
-Django==4.0
+Django==1.11.29
```
Curious why we need a `requirements.txt` in addition to the `pyproject.toml` + `poetry.lock`?

bluedenim replied:
We don't need it. It's leftover.

setu4993 reviewed (approved):
Thanks, Van! Left 1 note about using `django.conf.settings.DEFAULT_DB_ALIAS` instead of `django.db.DEFAULT_DB_ALIAS` to correctly match the default behavior with the docs. Rest looks great.

Thanks for cleaning up and making it nicer! Don't forget to version bump at the end!

bluedenim committed:
Changes from feedback: correction of source for DEFAULT_DB_ALIAS
Additional tweaks to provide convenient access to the connection_name parameter
Setting min Django version to 1.11.17 since that's the first version supporting Python 3.7

bluedenim committed:
Bump version to 0.2.0

bluedenim committed:
Update poetry.lock

setu4993 commented on vmigration_helper/helpers/command.py:
```diff
@@ -0,0 +1,51 @@
+        parser.add_argument(
+            '--connection-name',
+            default=DEFAULT_DB_ALIAS,
+            help=f'The connection to use for migration commands. Defaults to settings.DEFAULT_DB_ALIAS'
```
Nice, I like this clarification.

setu4993 reviewed (approved):
Looks great, thanks Van!

bluedenim committed:
Hunt down last place where we errorneously said
settings.DEFAULT_DB_ALIAS
